includes/REST/OrderControllerV2.php

```diff
@@ -0,0 +1,233 @@
+        $data      = [];
+        $downloads = [];
+
+        $download_permissions = $wpdb->get_results(
+            $wpdb->prepare(
+                "
+                SELECT * FROM {$wpdb->prefix}woocommerce_downloadable_product_permissions
+                WHERE order_id = %d ORDER BY product_id
+            ", $requests->get_param( 'id' )
+            )
+        );
+
+        $product = null;
+        $loop    = 0;
+
+        if ( $download_permissions && sizeof( $download_permissions ) > 0 ) {
+			foreach ( $download_permissions as $download ) {
+				if ( ! $product || dokan_get_prop( $product, 'id' ) != $download->product_id ) {
+					$product = wc_get_product( absint( $download->product_id ) );
+					$file_count = 0;
+				}
+
+				// don't show permissions to files that have since been removed
+				if ( ! $product || ! $product->exists() || ! $product->has_file( $download->download_id ) ) {
+					continue;
+				}
+
+				$downloads[] = $download;
+
+				$loop++;
+				$file_count++;
+			}
+        }
+
+        $data['downloads'] = $downloads;
+        $orders_items = wc_get_order( $requests->get_param( 'id' ) )->get_items();
+        $orders_items_ids = [];
+        foreach ( $orders_items as $item ) {
+            $orders_items_ids[] = $item->get_product_id();
+        }
+        $orders_items_ids = implode( ',', $orders_items_ids );
+
+        $products = $wpdb->get_results(
+            $wpdb->prepare(
+                "SELECT $wpdb->posts.* FROM $wpdb->posts
+                        INNER JOIN $wpdb->postmeta
+                            ON ( $wpdb->posts.ID = $wpdb->postmeta.post_id )
+                        WHERE $wpdb->posts.post_author=%d
+                            AND ( $wpdb->postmeta.meta_key = '_downloadable' AND $wpdb->postmeta.meta_value = 'yes' )
+                            AND $wpdb->posts.post_type IN ( 'product', 'product_variation' )
+                            AND $wpdb->posts.post_status = 'publish'
+                            AND $wpdb->posts.ID IN ( {$orders_items_ids} )
+                        GROUP BY $wpdb->posts.ID
+                        ORDER BY $wpdb->posts.post_parent ASC, $wpdb->posts.post_title ASC", $user_id
+            )
+        );
+
+        if ( $products ) {
+			foreach ( $products as $product ) {
+				$product_object = wc_get_product( $product->ID );
+				$product_name   = $product_object->get_formatted_name();
+
+				$data['products'][ $product->ID ] = esc_html( $product_name );
+			}
+        }
+
+        return rest_ensure_response( $data );
+    }
+
+    /**
+     * Grant downloadable product access to the given order.
+     *
+     * @param $requests
+     * @return WP_Error|\WP_HTTP_Response|\WP_REST_Response
+     */
+    public function grant_order_downloads( $requests ) {
+        $order_id     = intval( $requests->get_param( 'id' ) );
+        $product_ids  = array_filter( array_map( 'absint', (array) wp_unslash( $requests->get_param( 'ids' ) ) ) );
+        $loop         = 0;
```

Comment:
This variable is not doing actually anything as I'm seeing.
and also in L-196 is not needed.